src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,164 @@
+    private static RegistrationServiceImpl registrationService;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        testUser = new User();
+        testUser.setId(FIRST_TEST_ID);
+        testUser.setAge(AGE_18_YEARS);
+        testUser.setLogin(LOGIN_OK);
+        testUser.setPassword(PASSWORD_8_SYMBOL_OK);
+
+        secondTestUser = new User();
+        secondTestUser.setId(SECOND_TEST_ID);
+        secondTestUser.setAge(AGE_MORE_18_YEARS);
+        secondTestUser.setLogin(LOGIN_SECOND_OK);
+        secondTestUser.setPassword(PASSWORD_8_SYMBOL_SECOND_OK);
+
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_NotOk() {
+        testUser = null;
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "User must be NOT null");
+    }
+
+    @Test
+    void register_less18YearsAge_NotOk() {
+        testUser.setAge(AGE_LES_18_YEARS);
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "Age must be more then 18");
+    }
+
+    @Test
+    void register_nullAge_NotOk() {
+        testUser.setAge(null);
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "Age must be more NOT null");
+    }
+
+    @Test
+    void register_more_Then80_Age_NotOk() {
+        testUser.setAge(AGE_MORE_80_YEARS);
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "Age must be less then 80");
+    }
+
+    @Test
+    void register_loginContainsSpace_NotOk() {
+        testUser.setLogin(LOGIN_CONTAINS_SPACE);
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "Login must NOT contains space");
+    }
+
+    @Test
+    void register_login_null_NotOk() {
+        testUser.setLogin(null);
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "Login must NOT be null");
+    }
+
+    @Test
+    void register_password_null_NotOk() {
+        testUser.setPassword(null);
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "Password must NOT be null");
+    }
+
+    @Test
+    void register_password_LessThen_8_NotOk() {
+        testUser.setPassword(PASSWORD_8_SYMBOL_NOTOK);
+        assertThrows(RegisterValidationException.class, () -> {
+            registrationService.register(testUser);
+        }, "Password must be mo then 8 characters");
+    }
+
+    @Test
+    void register_login_OlreadyExist_NotOk() {
+        registrationService.register(testUser);
```

Comment:
In each test case when you test some behavior with some already existing user, we should not put him there using the same register method which is under test right now, and you cannot guarantee that it works as intended.
Instead, we should put this test data directly to `Storage` like `Storage.add(user)` or as close to the destination as possible, the lower the better.